batch/batch/worker/worker.py

```diff
@@ -3048,13 +3048,41 @@ async def create_job_1(self, request):
 
         batch_id = body['batch_id']
         job_id = body['job_id']
+        id = (batch_id, job_id)
 
         format_version = BatchFormatVersion(body['format_version'])
 
         token = body['token']
         start_job_id = body['start_job_id']
         addtl_spec = body['job_spec']
 
+        user = body['user']
+        gsa_key = body['gsa_key']
+
+        # already running
+        if id in self.jobs:
+            return web.HTTPForbidden()
```

Comment:
I worry that this is introducing a minor concurrency bug because the `if id in self.jobs` and the `self.jobs[job.id] = job` are no longer executed in the same synchronous block. A transient network error could mean that a second request to this endpoint gets past this check before the first request's task adds the job to `self.jobs`, resulting in two of the same job running on the machine.

There's lots of ways to fix this. Add another variant of `Job` that is a stand-in until the actual `Job` object can get resolved, make `self.jobs` a `Dict[Tuple[int, int], Future[Job]]`, acquire an `asyncio.Lock` that is released when the `Job` is added to the dict, etc.